.gitlab-ci.yml

```
@@ -213,47 +213,41 @@ deploy_artifacts_to_github:
     - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-java-agent/build/libs/*.jar
     - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-trace-api/build/libs/*.jar
     - gh release upload --clobber --repo DataDog/dd-trace-java $CI_COMMIT_TAG workspace/dd-trace-ot/build/libs/*.jar
-    
-deploy_to_docker_registries:
+
+generate-lib-init-tag-values:
+  tags: ["arch:amd64"]
+  image: registry.ddbuild.io/ci/auto_inject/gitlab:current
   stage: deploy
   rules:
-    - if: '$CI_COMMIT_TAG =~ /.*-[Rr][Cc].*/'
-      when: manual
-      allow_failure: true
     - if: '$POPULATE_CACHE'
       when: never
-    - if: '$CI_COMMIT_TAG =~ /^v.*/'
+    # We don't tag prerelease versions
+    - if: '$CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+$/'
       when: on_success
     - when: manual
       allow_failure: true
-  trigger:
-    project: DataDog/public-images
-    branch: main
-    strategy: depend
   variables:
-    IMG_SOURCES: ghcr.io/datadog/dd-trace-java/dd-lib-java-init:$CI_COMMIT_SHA
-    IMG_DESTINATIONS: dd-lib-java-init:$CI_COMMIT_TAG
-    IMG_SIGNING: "false"
+    IMG_DESTINATION_BASE: dd-lib-java-init
+  script:
+    - ./.gitlab/build-lib-init.sh
+  artifacts:
+    reports:
+      dotenv: build.env
 
-deploy_latest_tag_to_docker_registries:
+deploy-lib-init-trigger:
   stage: deploy
-  rules:
-    - if: '$CI_COMMIT_TAG =~ /.*-[Rr][Cc].*/'
-      when: manual
-      allow_failure: true
-    - if: '$POPULATE_CACHE'
-      when: never
-    - if: '$CI_COMMIT_TAG =~ /^v1\..*/'
-      when: on_success
-    - when: manual
-      allow_failure: true
+  # needs the version from the generate-tag-values job
+  needs:
+    - job: generate-lib-init-tag-values
+      artifacts: true
   trigger:
+#    project: DataDog/dd-trace-dotnet-gitlab-test # can be used for testing
     project: DataDog/public-images
     branch: main
     strategy: depend
   variables:
-    IMG_SOURCES: ghcr.io/datadog/dd-trace-java/dd-lib-java-init:$CI_COMMIT_SHA
-    IMG_DESTINATIONS: dd-lib-java-init:latest
+    IMG_SOURCES: ghcr.io/datadog/dd-trace-java/dd-lib-java-init:$CI_COMMIT_TAG
```

I think this was a typo, the _source_ of the image shouldn't have changed 
```suggestion
    IMG_SOURCES: ghcr.io/datadog/dd-trace-java/dd-lib-java-init:$CI_COMMIT_TAG
```
```suggestion
    IMG_SOURCES: ghcr.io/datadog/dd-trace-java/dd-lib-java-init:$CI_COMMIT_SHA
```